src/main.ts

```diff
@@ -98,28 +99,20 @@ const checkIfValidCLIVersion = async (azcliversion: string): Promise<boolean> =>
 }
 
 const getAllAzCliVersions = async (): Promise<Array<string>> => {
-    var outStream: string = '';
-    var execOptions: any = {
-        outStream: new NullOutstreamStringWritable({ decodeStrings: false }),
-        listeners: {
-            stdout: (data: any) => outStream += data.toString() + os.EOL, //outstream contains the list of all the az cli versions
-        }
-    };
-
     try {
-        await exec.exec("curl", ["--location", "-s", "https://mcr.microsoft.com/v2/azure-cli/tags/list"], execOptions)
-        if (outStream && JSON.parse(outStream).tags) {
-            return JSON.parse(outStream).tags;
+        const response = await axios.get('https://mcr.microsoft.com/v2/azure-cli/tags/list');

```

Comment:
Since this action runs on Node 20, native [fetch](https://nodejs.org/docs/v20.15.1/api/globals.html#fetch) is available and wouldn't require an additional dependency.

Reply:
I am happy to try it out again from a branch on my workflow if that helps and you want to test the change